ideafast/middleware-services
# Integrate DMP upload from dmpy

## What Changed

Use the modified `dmpy` client to upload a file to the DMP. I've also added logging to show logs output from`dmpy.client` and that we can use elsewhere (i.e., #31). This resolves #30.

## Testing

1. Create a local file named `dmpy.env` and add the relevant credentials.
2. Download `51eb52d7dd8d5c6570e123f48b3d05898940aaf744ee68cfa9fdd1338237dfa2` file from the 'hackathon_test' dataset (ideally the dreem file that is 2.2GB) and store it in `/data/uploading`
3. Rename this file to `NXQEXRT-DRMHK9K4K-20210121-20210125` and make sure there are no other files in the upload folder.
4. As we only want to test the upload feature, comment all other pipeline stages in  `main.py`, i.e., [here](https://github.com/ideafast/middleware-services/blob/master/data_transfer/main.py#L45-L57)
5. Also comment out all lines in [`batch_upload_data` besides the upload](https://github.com/ideafast/middleware-services/blob/master/data_transfer/jobs/shared.py#L17-L28).
5. Run `python data_transfer/main.py` and observe the log output. Upload make take some time.
6. Once uploaded, verify on the DMP that the file was uploaded. 

Note: please review this in tandem with [upload implementation in dmpy](https://github.com/ideafast/dmpy/pull/2).
